valhalla/baldr/time_info.h

```diff
@@ -72,6 +73,35 @@ struct TimeInfo {
         break;
     }
 
+    // return the time info based on this location information
+    return make(*location.mutable_date_time(), timezone_index, tz_cache, default_timezone_index);
+  }
+
+  /**
+   * Helper function to initialize the object from a location. Uses the graph to
+   * find timezone information about the edge candidates at the location. If the
+   * graph has no timezone information or the location has no edge candidates the
+   * default timezone will be used (if unspecified UTC is used). If no datetime
+   * is provided on the location or an invalid one is provided the TimeInfo will
+   * be invalid.
+   *
+   * @param date_time               string representing the time from which to make this time info
+   *                                if the string is 'current' it will be replaced with the now time
+   * @param timezone_index          the timezone index at which to make this time info, 0 if unknown
+   * @param tz_cache                timezone info cache used when crossing timezones while tracking
+   * @param default_timezone_index  the index to use when the timezone index is 0 or unknown
+   * @return the initialized TimeInfo
+   */
+  static TimeInfo
+  make(std::string& date_time,
```

Comment:
i had to break this out into 2 functions so that the recosting could make use of it